
--- a/projectile-rails.el
+++ b/projectile-rails.el
@@ -1812,70 +1812,6 @@
                      ("x" "extract region" projectile-rails-extract-region))))
    :bind "")) ;;accessible only from the main context menu
 
-(with-no-warnings
-  (ignore-errors
-    (defhydra hydra-projectile-rails-find (:color blue :columns 8)
-      "Find a resources"
-      ("m" projectile-rails-find-model       "model")
-      ("v" projectile-rails-find-view        "view")
-      ("c" projectile-rails-find-controller  "controller")
-      ("h" projectile-rails-find-helper      "helper")
-      ("l" projectile-rails-find-lib         "lib")
-      ("j" projectile-rails-find-javascript  "javascript")
-      ("w" projectile-rails-find-component  "component")
-      ("s" projectile-rails-find-stylesheet  "stylesheet")
-      ("p" projectile-rails-find-spec        "spec")
-      ("u" projectile-rails-find-fixture     "fixture")
-      ("t" projectile-rails-find-test        "test")
-      ("f" projectile-rails-find-feature     "feature")
-      ("i" projectile-rails-find-initializer "initializer")
-      ("o" projectile-rails-find-log         "log")
-      ("@" projectile-rails-find-mailer      "mailer")
-      ("!" projectile-rails-find-validator   "validator")
-      ("y" projectile-rails-find-layout      "layout")
-      ("n" projectile-rails-find-migration   "migration")
-      ("k" projectile-rails-find-rake-task   "rake task")
-      ("b" projectile-rails-find-job         "job")
-      ("z" projectile-rails-find-serializer  "serializer")
-
-      ("M" projectile-rails-find-current-model      "current model")
-      ("V" projectile-rails-find-current-view       "current view")
-      ("C" projectile-rails-find-current-controller "current controller")
-      ("H" projectile-rails-find-current-helper     "current helper")
-      ("J" projectile-rails-find-current-javascript "current javascript")
-      ("S" projectile-rails-find-current-stylesheet "current stylesheet")
-      ("P" projectile-rails-find-current-spec       "current spec")
-      ("U" projectile-rails-find-current-fixture    "current fixture")
-      ("T" projectile-rails-find-current-test       "current test")
-      ("N" projectile-rails-find-current-migration  "current migration")
-      ("Z" projectile-rails-find-current-serializer "current serializer"))
-
-    (defhydra hydra-projectile-rails-goto (:color blue :columns 8)
-      "Go to"
-      ("f" projectile-rails-goto-file-at-point "file at point")
-      ("g" projectile-rails-goto-gemfile       "Gemfile")
-      ("p" projectile-rails-goto-package       "package")
-      ("r" projectile-rails-goto-routes        "routes")
-      ("d" projectile-rails-goto-schema        "schema")
-      ("s" projectile-rails-goto-seeds         "seeds")
-      ("h" projectile-rails-goto-spec-helper   "spec helper"))
-
-    (defhydra hydra-projectile-rails-run (:color blue :columns 8)
-      "Run external command & interact"
-      ("r" projectile-rails-rake       "rake")
-      ("c" projectile-rails-console    "console")
-      ("b" projectile-rails-dbconsole  "dbconsole")
-      ("s" projectile-rails-server     "server")
-      ("g" projectile-rails-generate   "generate")
-      ("d" projectile-rails-destroy    "destroy")
-      ("x" projectile-rails-extract-region "extract region"))
-
-    (defhydra hydra-projectile-rails (:color blue :columns 8)
-      "Projectile Rails"
-      ("f" hydra-projectile-rails-find/body "Find a resource")
-      ("g" hydra-projectile-rails-goto/body "Goto")
-      ("r" hydra-projectile-rails-run/body "Run & interact"))))
-
 (provide 'projectile-rails)
 
 ;;; projectile-rails.el ends here

