--- a/usr/src/blackmagic-io-12.4.1a15/bm_util.c	2023-03-16 21:21:13.731668636 +0100
+++ b/usr/src/blackmagic-io-12.4.1a15/bm_util.c	2023-03-16 21:23:48.468890382 +0100
@@ -902,7 +902,11 @@
 // Random
 uint32_t bm_random32(void)
 {
-#if KERNEL_VERSION_OR_LATER(3, 8, 0)
+#if KERNEL_VERSION_OR_LATER(6, 1, 0)
+	// INOFFICIAL PATCH: prandom_u32() got removed in Linux kernel 6.1, it was a alias to get_random_u32() before
+	// see: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.19&id=de492c83cae0af72de370b9404aacda93dafcad5
+	return get_random_u32();
+#elif KERNEL_VERSION_OR_LATER(3, 8, 0)
 	return prandom_u32();
 #elif KERNEL_VERSION_OR_LATER(2, 6, 19)
 	return random32();
