diff --git a/supergameboy/Makefile b/supergameboy/Makefile
index 60409af..cbef9ac 100755
--- a/supergameboy/Makefile
+++ b/supergameboy/Makefile
@@ -2,14 +2,13 @@ include nall/Makefile
 
 c     := $(compiler) -x c -std=gnu99
 cpp   := $(compiler) -std=gnu++0x
-flags := -O3 -fomit-frame-pointer -I. -Icommon -Ilibgambatte/include -Ilibgambatte/src
+flags := $(CXXFLAGS) -fomit-frame-pointer -I. -Icommon -Ilibgambatte/include -Ilibgambatte/src
 link  :=
 
 ifeq ($(platform),osx)
   flags := -fPIC $(flags)
 else ifeq ($(platform),x)
   flags := -fPIC $(flags)
-  link += -s
 endif
 
 objects := supergameboy
