diff -rupN terminator-2.1.2.old/terminatorlib/terminal.py terminator-2.1.2/terminatorlib/terminal.py
--- terminator-2.1.2.old/terminatorlib/terminal.py	2022-10-19 16:22:21.000000000 +0200
+++ terminator-2.1.2/terminatorlib/terminal.py	2023-01-13 08:16:57.109231720 +0100
@@ -910,6 +910,20 @@ class Terminal(Gtk.VBox):
         # FIXME: Does keybindings really want to live in Terminator()?
         mapping = self.terminator.keybindings.lookup(event)
 
+        # Just propagate tab-swictch events if there is only one tab
+        if (
+            mapping and (
+                mapping.startswith('switch_to_tab')
+                or mapping in ('next_tab', 'prev_tab')
+            )
+        ):
+            window = self.get_toplevel()
+            child = window.get_children()[0]
+            if isinstance(child, Terminal):
+                # not a Notebook instance => a single tab is used
+                # .get_n_pages() can not be used
+                return False
+
         if mapping == "hide_window":
             return False
 
