diff -rupN sylpheed-3.7.0/libsylph/utils.c sylpheed-3.7.0.new/libsylph/utils.c
--- sylpheed-3.7.0/libsylph/utils.c	2017-11-30 03:34:14.000000000 +0100
+++ sylpheed-3.7.0.new/libsylph/utils.c	2018-02-08 08:39:06.586381290 +0100
@@ -3306,10 +3306,10 @@ gint canonicalize_file(const gchar *src,
 		}
 	}
 
-	if (last_linebreak == TRUE) {
-		if (fputs("\r\n", dest_fp) == EOF)
-			err = TRUE;
-	}
+	/* if (last_linebreak == TRUE) { */
+	/* 	if (fputs("\r\n", dest_fp) == EOF) */
+	/* 		err = TRUE; */
+	/* } */
 
 	if (ferror(src_fp)) {
 		FILE_OP_ERROR(src, "fgets");
