diff --git a/gucharmap/gucharmap-unicode-info.h b/gucharmap/gucharmap-unicode-info.h
index 5b086c285..9d07c5cfd 100644
--- a/gucharmap/gucharmap-unicode-info.h
+++ b/gucharmap/gucharmap-unicode-info.h
@@ -41,7 +41,20 @@ typedef enum {
   GUCHARMAP_UNICODE_VERSION_5_1,
   GUCHARMAP_UNICODE_VERSION_5_2,
   GUCHARMAP_UNICODE_VERSION_6_0,
-  GUCHARMAP_UNICODE_VERSION_LATEST = GUCHARMAP_UNICODE_VERSION_6_0 /* private, will move forward with each revision */
+  GUCHARMAP_UNICODE_VERSION_6_1,
+  GUCHARMAP_UNICODE_VERSION_6_2,
+  GUCHARMAP_UNICODE_VERSION_6_3,
+  GUCHARMAP_UNICODE_VERSION_7_0,
+  GUCHARMAP_UNICODE_VERSION_8_0,
+  GUCHARMAP_UNICODE_VERSION_9_0,
+  GUCHARMAP_UNICODE_VERSION_10_0,
+  GUCHARMAP_UNICODE_VERSION_11_0,
+  GUCHARMAP_UNICODE_VERSION_12_0,
+  GUCHARMAP_UNICODE_VERSION_12_1,
+  GUCHARMAP_UNICODE_VERSION_13_0,
+  GUCHARMAP_UNICODE_VERSION_14_0,
+  GUCHARMAP_UNICODE_VERSION_15_0,
+  GUCHARMAP_UNICODE_VERSION_LATEST = GUCHARMAP_UNICODE_VERSION_15_0 /* private, will move forward with each revision */
 } GucharmapUnicodeVersion;
 
 /* return values are read-only */
