From 09d185d27bc31741779022ff73b199d9b14ed760 Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Fri, 10 Feb 2023 16:16:21 +0100
Subject: [PATCH] Add missing <list> include

--- a/source/uiApp.cpp
+++ b/source/uiApp.cpp
@@ -66,6 +66,7 @@
 #include <string.h>
 #include <string>
 #include <sstream>
+#include <list>
 #include <map>
 #include <vector>
 #include <algorithm>
-- 
2.39.1

