diff --git a/src/remoteobjects/qconnectionfactories.cpp b/src/remoteobjects/qconnectionfactories.cpp
index 1dc6cfc..b7b9de8 100644
--- a/src/remoteobjects/qconnectionfactories.cpp
+++ b/src/remoteobjects/qconnectionfactories.cpp
@@ -234,7 +234,7 @@ ExternalIoDevice::ExternalIoDevice(QIODevice *device, QObject *parent)
     connect(m_device.data(), &QIODevice::aboutToClose, this, [this]() { this->m_isClosing = true; });
     connect(m_device.data(), &QIODevice::readyRead, this, &ExternalIoDevice::readyRead);
     auto meta = device->metaObject();
-    if (-1 != meta->indexOfSignal(SIGNAL(disconnected())))
+    if (-1 != meta->indexOfSignal("disconnected()"))
       connect(m_device.data(), SIGNAL(disconnected()), this, SIGNAL(disconnected()));
 }
 
