Fix compilation with glibc-2.28.
https://bugs.gentoo.org/681904

--- emacs-24.3-orig/configure.ac
+++ emacs-24.3/configure.ac
@@ -3597,7 +3597,8 @@
       dnl since the preprocessed output is accessible in "conftest.i".
       AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
 #include <stdio.h>
-#if !defined (_IO_STDIO_H) && !defined (_STDIO_USES_IOSTREAM)
+#if !defined (_IO_STDIO_H) && !defined (_STDIO_USES_IOSTREAM) \
+    && !defined _IO_EOF_SEEN && __GNU_LIBRARY__ != 1
 # error "stdio definitions not found"
 #endif
         ]], [[]])], emacs_pending_output=new, [])
