Use local Elisp dependencies.

--- a/docs/Makefile
+++ b/docs/Makefile
@@ -1,4 +1,8 @@
-EMACS=emacs -Q -batch -f package-initialize -L ../ -L ../../ellit-org.el -l ellit-org
+SITELISP = /usr/share/emacs/site-lisp
+BUILDDEPS = alert all-the-icons dashboard compat ellit-org esxml htmlize org \
+            rainbow-identifiers transient visual-fill-column
+BUILDDEPS:=$(BUILDDEPS:%=-L $(SITELISP)/%)
+EMACS=emacs -Q -batch -L ../ $(BUILDDEPS) -l ellit-org
 
 DOC_EL_SOURCES=../telega.el ../telega-*.el ../contrib/*.el
 

Use local setupfile.

--- a/docs/index-html.org
+++ b/docs/index-html.org
@@ -1,6 +1,6 @@
 #+EXPORT_FILE_NAME: index.html
 
-#+SETUPFILE: https://fniessen.github.io/org-html-themes/org/theme-readtheorg.setup
+#+SETUPFILE: theme-readtheorg.setup
 #+HTML_HEAD: <style> #content{max-width:1800px;}</style>
 #+HTML_HEAD: <style>h1, h2, h3, h4, h5, h6 { margin: 16px 0px 0px 0px; }</style>
 #+HTML_HEAD: <style>#table-of-contents h2 { margin: 0; }</style>

