diff --git a/snesfilter/Makefile b/snesfilter/Makefile
index 089b86e..08ba81b 100755
--- a/snesfilter/Makefile
+++ b/snesfilter/Makefile
@@ -5,12 +5,12 @@ include nall/qt/Makefile
 
 c     := $(compiler) -std=gnu99
 cpp   := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags := -O3 -I. -Iobj -fomit-frame-pointer $(qtinc)
+flags := $(CXXFLAGS) -I. -Iobj -fomit-frame-pointer $(qtinc)
 link  :=
 
 ifeq ($(platform),x)
   flags := -fPIC -fopenmp $(flags)
-  link += -s -fopenmp -lpthread -lgomp
+  link += -fopenmp -lpthread -lgomp
 else ifeq ($(platform),osx)
   flags := -fPIC -fopenmp $(flags)
   link += -fopenmp -lpthread -lgomp
@@ -73,9 +73,8 @@ endif
 
 install:
 ifeq ($(platform),x)
-	install -D -m 755 libsnesfilter.a $(DESTDIR)$(prefix)/lib
-	install -D -m 755 libsnesfilter.so $(DESTDIR)$(prefix)/lib
-	ldconfig -n $(DESTDIR)$(prefix)/lib
+	install -D -m 755 libsnesfilter.a $(DESTDIR)$(prefix)/lib/libsnesfilter.a
+	install -D -m 755 libsnesfilter.so $(DESTDIR)$(prefix)/lib/libsnesfilter.so
 else ifeq ($(platform),osx)
 	cp libsnesfilter.dylib /usr/local/lib/libsnesfilter.dylib
 endif
