diff --git a/Makefile b/Makefile
index 0d6280c..83e9505 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ link  :=
 ifeq ($(DEBUG), 1)
   flags += -O0 -g
 else
-  flags += -O3 -fomit-frame-pointer
+  flags += $(CXXFLAGS) -fomit-frame-pointer
 endif
 
 # silence warnings
@@ -19,7 +19,7 @@ flags += -Wno-switch -Wno-absolute-value -Wno-parentheses
 
 ifeq ($(platform),x)
   flags := -fPIC -fopenmp $(flags)
-  link += -s -fopenmp -lpthread -lgomp
+  link += -fopenmp -lpthread -lgomp
 else ifeq ($(platform),osx)
 # TODO: openmp not currently supported in Apple clang
 #  flags := -fPIC -fopenmp $(flags)
