Remove all mentions of emms

--- a/elfeed-show.el
+++ b/elfeed-show.el
@@ -72,9 +72,7 @@
       (define-key map "\e\t" #'shr-previous-link)
       (define-key map [backtab] #'shr-previous-link)
       (define-key map "c" #'elfeed-kill-link-url-at-point)
-      (define-key map [mouse-2] #'shr-browse-url)
-      (define-key map "A" #'elfeed-show-add-enclosure-to-playlist)
-      (define-key map "P" #'elfeed-show-play-enclosure)))
+      (define-key map [mouse-2] #'shr-browse-url)))
   "Keymap for `elfeed-show-mode'.")
 
 (defun elfeed-show-mode ()
@@ -434,27 +432,6 @@
       1
     (elfeed--get-enclosure-num "Enclosure to play" entry)))
 
-(defun elfeed-show-play-enclosure (enclosure-index)
-  "Play enclosure number ENCLOSURE-INDEX from current entry using EMMS.
-Prompts for ENCLOSURE-INDEX when called interactively."
-  (interactive (list (elfeed--enclosure-maybe-prompt-index elfeed-show-entry)))
-  (elfeed-show-add-enclosure-to-playlist enclosure-index)
-  (with-no-warnings
-    (with-current-emms-playlist
-      (save-excursion
-        (emms-playlist-last)
-        (emms-playlist-mode-play-current-track)))))
-
-(defun elfeed-show-add-enclosure-to-playlist (enclosure-index)
-  "Add enclosure number ENCLOSURE-INDEX to current EMMS playlist.
-Prompts for ENCLOSURE-INDEX when called interactively."
-
-  (interactive (list (elfeed--enclosure-maybe-prompt-index elfeed-show-entry)))
-  (require 'emms) ;; optional
-  (with-no-warnings ;; due to lazy (require )
-    (emms-add-url   (car (elt (elfeed-entry-enclosures elfeed-show-entry)
-                              (- enclosure-index 1))))))
-
 (defun elfeed-show-next-link ()
   "Skip to the next link, exclusive of the Link header."
   (interactive)
