diff --git a/Makefile b/Makefile
index 497f591..ad6e758 100644
--- a/Makefile
+++ b/Makefile
@@ -34,11 +34,11 @@ endif
 ifeq ($(DEBUG), 1)
   flags += -O0 -g
 else
-  flags += -O3 -fomit-frame-pointer
+  flags += $(CXXFLAGS) -fomit-frame-pointer
   ifeq ($(platform),osx)
     # -s is deprecated as of Apple LLVM 7.0, and using it breaks linking with some versions
   else
-    link += -s
+    # link += -s
   endif
 endif
 
@@ -107,7 +107,6 @@ ifeq ($(platform),x)
 	install -D -m 755 out/$(out) $(DESTDIR)$(prefix)/bin/$(out)
 	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
 	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
-	test -d ~/.bsnes || mkdir ~/.bsnes
 else ifeq ($(platform),osx)
 	mv -f $(osxbundle) /Applications/bsnes+.app
 	test -d ~/.bsnes || mkdir ~/.bsnes
